### PR TITLE
Refine Mixpanel click logging for radio inputs (SCP-3081)

### PR DESCRIPTION
**Pull request by eweitz**
This provides useful Mixpanel logging for clicks on radio-type inputs, as seen in e.g. expression matrix upload UI.

Previously, such clicks were logged but had essentially worthless analytics: `label` was empty and no specific information was provided beyond "there was a radio button click on this page".  Now, we record the `id`, `input-name`, and `value` of the selected radio button option.  

That allows us to analyze things like "What do users do after clicking 'Yes' for 'Is this a raw counts matrix'?"

Due to the complicated nature of radio inputs and how we use them, the raw HTML attribute values are logged rather than their human-friendly labels.  So radio button analysis will need a bit of effort, but is now at least possible -- and the instrumentation is easy to maintain.

To test:
* Go to the Upload Wizard for a study.
* Open Network panel in DevTools, filter to "bard method:POST".
* In the "Expression" tab, click a radio button.
* Verify that an event is sent to Bard with non-empty values for `id`, `input-name`, and `value`.

This is needed for SCP-3081.

### CODE REVIEW

**Review by bistline — APPROVED** [review]

File: app/javascript/lib/metrics-api.js [hunk 1]
```diff
@@ -163,17 +163,24 @@ function getLabelsForElement(element) {
  * Log click on input by type, e.g. text, number, checkbox
  */
 function logClickInput(target) {
-  const domLabels = getLabelsForElement(target)
+  let props
+  if (target.type === 'radio') {
+    const id = 'id' in target ? target.id : ''
```
Comment (devonbush) [inline app/javascript/lib/metrics-api.js:168]: I assume you tested this and so it's there for a reason, but I thought (element).id already returned empty string if no id was specified for the element

**Review by devonbush — APPROVED** [review]
Looks good, one html-ese question

File: app/javascript/lib/metrics-api.js (hunk 1 above)
Comment (eweitz) [reply app/javascript/lib/metrics-api.js]: Good catch, refined in f86f76a.  I used this construct yesterday for something else, then removed it, then reflexively used it here.

**Comment by eweitz** [comment]
The [build failure](https://travis-ci.com/github/broadinstitute/single_cell_portal_core/builds/221293524#L3275) is a false positive:

```
Finished in 1274.627726s, 0.2016 runs/s, 0.9281 assertions/s.
  1) Failure:
FileParseServiceTest#test_should_parse_gene_list_file [/home/app/webapp/test/integration/lib/file_parse_service_test.rb:146]:
Expected false to be truthy.
257 runs, 1183 assertions, 1 failures, 0 errors, 1 skips
```

No Rails code was changed, so the test failure is not caused by this change.

**Pull request merged**